Yosemite/Yosemite/Stores/CustomerStore.swift

```diff
@@ -104,42 +108,84 @@ public final class CustomerStore: Store {
     ///
     /// - Parameters:
     ///   - siteID: The site for which customers should be fetched.
+    ///   - keyword: The keyword used for the Customer search query.
     ///   - searchResults: A WCAnalyticsCustomer collection that represents the matches we've got from the API based in our keyword search.
     ///   - onCompletion: Invoked when the operation finishes. Will map the result to a `[Customer]` entity.
     ///
     private func mapSearchResultsToCustomerObjects(for siteID: Int64,
+                                                   with keyword: String,
                                           with searchResults: [WCAnalyticsCustomer],
                                                   onCompletion: @escaping (Result<[Customer], Error>) -> Void) {
-        var results = [Customer]()
+        var customers = [Customer]()
         let group = DispatchGroup()
         for result in searchResults {
             group.enter()
             self.retrieveCustomer(for: siteID, with: result.userID, onCompletion: { result in
                 if let customer = try? result.get() {
-                    results.append(customer)
+                    customers.append(customer)
                 }
                 group.leave()
             })
         }
 
         group.notify(queue: .main) {
-            self.upsertSearchCustomerResults(siteID: siteID, readOnlySearchResults: searchResults, onCompletion: {})
-            onCompletion(.success(results))
+            self.upsertSearchCustomerResult(
+                siteID: siteID,
+                keyword: keyword,
+                readOnlyCustomers: customers,
+                onCompletion: {
+                    onCompletion(.success(customers))
+                }
+            )
         }
     }
+}
 
+// MARK: Storage operations
+private extension CustomerStore {
     /// Inserts or updates CustomerSearchResults in Storage
     ///
-    private func upsertSearchCustomerResults(siteID: Int64, readOnlySearchResults: [Networking.WCAnalyticsCustomer], onCompletion: @escaping () -> Void) {
-        for _ in readOnlySearchResults {
-            // Logic for inserting or updating in Storage will go here. Not implemented yet.
-            // https://github.com/woocommerce/woocommerce-ios/issues/7741
+    private func upsertSearchCustomerResult(siteID: Int64,
+                                            keyword: String,
+                                            readOnlyCustomers: [Networking.Customer],
+                                            onCompletion: @escaping () -> Void) {
+        sharedDerivedStorage.perform {
```

Comment:
Since we're using `self.sharedDerivedStorage` in the closure here, it would be good to capture `weak` self, rather than implicitly capturing a strong reference to self. (Another option is to assign `sharedDerivedStorage` to a constant in the method before using it, so you don't need a reference to self in the closure.)